xchange-kraken/src/main/java/info/bitrich/xchangestream/kraken/KrakenStreamingMarketDataService.java

```diff
@@ -0,0 +1,171 @@
+                                    return KrakenAdapters.adaptTrade(
+                                            new KrakenPublicTrade(
+                                                    bd(tradeList, 0),
+                                                    bd(tradeList, 1),
+                                                    getValue(tradeList, 2, Double.class),
+                                                    type == null ? null : KrakenType.fromString(type),
+                                                    orderType == null ? null : KrakenOrderType.fromString(orderType),
+                                                    getValue(tradeList, 5, String.class)
+                                            ), currencyPair
+                                    );
+                                }
+                        ).collect(Collectors.toList())
+                ));
+    }
+
+    public Observable<List> subscribe(String channelName, int maxItems, Integer depth) {
+        return service.subscribeChannel(channelName, depth)
+                .filter(JsonNode::isArray)
+                .filter(Objects::nonNull)
+                .map(jsonNode -> StreamingObjectMapperHelper.getObjectMapper().treeToValue(jsonNode, List.class))
+                .filter(list -> {
+                    if (list.size() < maxItems) {
+                        LOG.error("Invalid message in channel {}. It contains {} array items but expected at least {}", channelName, list.size(), maxItems);
+                        return false;
+                    }
+                    return true;
+                });
+    }
+
+    private String getChannelName(KrakenSubscriptionName subscriptionName, CurrencyPair currencyPair) {
+        String pair = currencyPair.base.toString() + "/" + currencyPair.counter.toString();
+        return subscriptionName + KRAKEN_CHANNEL_DELIMITER + pair;
+    }
+
+    private BigDecimal bd(List list, int index) {
+        return getValue(list, index, BigDecimal.class);
+    }
+
+    @SuppressWarnings("unchecked")
+    private <T> T getValue(List list, int index, Class<T> clazz) {
+        if (list == null || list.size() < index + 1) {
+            return null;
+        }
+        return (T) ConvertUtils.convert(list.get(index), clazz);
+    }
+
+    private int parseOrderBookSize(Object[] args) {
+        if (args != null && args.length > 0) {
+            Object obSizeParam = args[0];
+            LOG.debug("Specified Kraken order book size: {}", obSizeParam);
+            if (Number.class.isAssignableFrom(obSizeParam.getClass())) {
+                int obSize = ((Number) obSizeParam).intValue();
+                if (ArrayUtils.contains(KRAKEN_VALID_ORDER_BOOK_SIZES, obSize)) {
+                    return obSize;
+                }
+                LOG.error("Invalid order book size {}. Valid values: {}. Default order book size has been used: {}",
```

Comment:
Would an exception not be better here?

Reply:
I dont' like exception here. There is not critical issue about not supported orderbook depth. I think library user will note either this error in the log or short orderbook depth and fix parameter to required.  Maybe it will be better to use warn log level - but "error" looks more noticeable from my perspective  :)

Reply:
Fine.